high_school/views.py

```diff
@@ -66,32 +66,77 @@ class HighSchoolListView(ListView):
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
+        self.user_type = None
         self.dbn = None
         self.query = None
+        self.loc_filter = {}
+
+    def get(self, *args, **kwargs):
+        if not self.request.session.get("is_login", None):
+            return redirect("landingpage:index")
+        return super(HighSchoolListView, self).get(*args, **kwargs)
 
     def get_queryset(self):
         if "dbn" in self.kwargs:
             self.dbn = self.kwargs["dbn"]
         self.query = self.request.GET.get("q")
-        """if query:
-            return HighSchool.objects.filter(school_name__icontains=query).order_by(
-                "school_name"
-            )  # noqa: E501
-        else:"""
+        self.loc_filter["X"] = self.request.GET.get("loc_bx")
```

Comment:
can you elaborate on this code?

Reply:
It gets the particular filter's value from the request parameter (as it is set by submitting the form when a user changes the status of checkbox - onchange). We use this value to form a list, against which the dB is queried for the particular value of borough in "Boro" column.

Reply:
Instead of string, can you directly store it as a single character in the same way in which it is stored in db so that u can directly filter it.

Reply:
It is a single character, but what can happen is the user can have more than 1 checkbox selected, in which case we'll need a str comparison (something like boro in "K, M, Q"). That's why there's a loop which goes over all the currently selected borough checkboxes and forms that string.

Reply:
You can use boro__in=["K","M"] in filter() which will filter all boro containing letters K or M. If this cannot be used in this case, then your code looks fine.

Reply:
Okay, so I think we both are talking about the same thing and that document is what I actually referred to for filtering, but what I'm trying to explain is the part "['K', 'M']" needs to be dynamic based on the checkbox selected in the UI.

Reply:
And that's the reason why there's a need for a dict type entry and a loop to get the selected checkboxes

Reply:
Okay we can keep this for now then